packages/components/src/range-control/styles/range-control-styles.ts

```diff
@@ -18,23 +18,39 @@ import type {
 	TooltipProps,
 	TrackProps,
 	WrapperProps,
+	RangeControlProps,
 } from '../types';
+import type { NumberControlProps } from '../../number-control/types';
 
 const rangeHeightValue = 30;
+const rangeHeightValueNext = 40;
 const railHeight = 4;
 const rangeHeight = () =>
 	css( { height: rangeHeightValue, minHeight: rangeHeightValue } );
 const thumbSize = 12;
 
-export const Root = styled.div`
+const rootMinHeight = ( {
+	__next40pxDefaultSize,
+}: Pick< RangeControlProps, '__next40pxDefaultSize' > ) =>
+	css( {
+		minHeight: __next40pxDefaultSize
+			? rangeHeightValueNext
+			: rangeHeightValue,
+	} );
+
+type RootProps = Pick< RangeControlProps, '__next40pxDefaultSize' >;
+export const Root = styled.div< RootProps >`
 	-webkit-tap-highlight-color: transparent;
-	align-items: flex-start;
+	align-items: center;
 	display: flex;
 	justify-content: flex-start;
 	padding: 0;
 	position: relative;
 	touch-action: none;
 	width: 100%;
+
+	/* TODO: remove after removing the __next40pxDefaultSize prop */
+	${ rootMinHeight }
```

Comment:
Just a stylistic thing, but as a courtesy for when we actually have to remove the temporary `__next*` props, I find it cleaner to structure the code so that the `! __next` case is already handled as the exception case.

For example:

```suggestion
	min-height: 40px;
	
	${ deprecatedHeight };
```

where `deprecatedHeight` only handles the `! __next40pxDefaultSize` case. The clean up will be easier this way. ([Actual example](https://github.com/WordPress/gutenberg/blob/33fbd4a4fcb550b9a5174fd92a3a2db63c0eaaac/packages/components/src/combobox-control/styles.ts))

Reply:
Apologies, I had actually suggested Chad to write code this way!

Reply:
Either works for me, happy to switch it back to the previous approach that matches your recommendation @mirka. da32f2d92d